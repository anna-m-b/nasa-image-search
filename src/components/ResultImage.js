import React from 'react';
import PropTypes from 'prop-types';
import '../styles/ResultImage.css';

const ResultImage = ({ url, description, linkToOriginal }) => {
  return (
    <div className="result-image">
      <a href={linkToOriginal}>
        <img
          className="result-image_img"
          src={url}
          alt={description}
          data-testid="result-image"
        />
      </a>
    </div>
  );
};

ResultImage.propTypes = {
  url: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  linkToOriginal: PropTypes.string.isRequired,
}

export default ResultImage;
