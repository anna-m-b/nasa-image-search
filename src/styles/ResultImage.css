.result-image {
  margin: 0.5rem;
  width: 100%;
  height: auto;
  display: inline-block;

}

.result-image_img {
  max-width: 100%;
  border-radius: 5px;
  
}