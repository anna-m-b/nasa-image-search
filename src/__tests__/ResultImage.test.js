import ResultImage from '../components/ResultImage';
import { render } from '@testing-library/react';

describe('ResultImage', () => {
  const validProps = {
    url: 'https://via.placeholder.com/200',
    description: 'test title',
    linkToOriginal: 'https://via.placeholder.com/500',
  };
  test('it renders correctly', () => {
    const { asFragment } = render(<ResultImage {...validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('it renders props correctly', () => {
    beforeEach(() => {
      render(<ResultImage {...validProps} />);
    });
    test('image src is set to url prop', () => {
      const image = document.querySelector('img');
      expect(image.src).toEqual(validProps.url);
    });
    test('image alt text is set to description prop', () => {
      const image = document.querySelector('img');
      expect(image.alt).toEqual(validProps.description);
    });
    test('anchor tag has its href set to linkToOriginal prop', () => {
      const anchor = document.querySelector('a');
      expect(anchor.href).toEqual(validProps.linkToOriginal);
    });

  });
});
