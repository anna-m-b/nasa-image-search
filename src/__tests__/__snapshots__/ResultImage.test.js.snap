// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultImage it renders correctly 1`] = `
<DocumentFragment>
  <div
    class="result-image"
  >
    <a
      href="https://via.placeholder.com/500"
    >
      <img
        alt="test title"
        class="result-image_img"
        data-testid="result-image"
        src="https://via.placeholder.com/200"
      />
    </a>
  </div>
</DocumentFragment>
`;
